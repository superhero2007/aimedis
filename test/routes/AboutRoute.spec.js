import React from 'react';
import Router from 'react-router-dom/MemoryRouter';
import { renderToString } from 'react-dom/server';
import AboutRoute from 'routes/AboutRoute';

describe('routes/AboutRoute', () => {
  it('should render the About us Page for unauthenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/about']}>
        <AboutRoute
          exact
          path="/about"
          component={() => (<div>About us</div>)}
          isAuthenticated={false}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });

  it('should render the About us Page for authenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/about']}>
        <AboutRoute
          exact
          path="/about"
          component={() => (<div>About us</div>)}
          isAuthenticated={true}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });
});
