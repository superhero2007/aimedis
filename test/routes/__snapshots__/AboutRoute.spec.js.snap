// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes/AboutRoute should render the About us Page for authenticated access 1`] = `"<div data-reactroot=\\"\\">About us</div>"`;

exports[`routes/AboutRoute should render the About us Page for unauthenticated access 1`] = `"<div data-reactroot=\\"\\">About us</div>"`;
